File: scripts/gen-private-post.js
#!/bin/node

import crypto from "node:crypto";
import { argv } from "node:process";
import fs from "node:fs";
import path from "node:path";

import moment from "moment";

import { getBlogpostData } from "../src/blog.js";

if (argv.length < 3) {
	console.log("err: provide blogpost id");
	process.exit(1);
}

const postPath = argv[2];

if (path.dirname(postPath) !== "private-blogposts") {
	console.log("err: post must be in `private-blogposts/`");
	process.exit(1);
}

if (!fs.existsSync(postPath)) {
	console.log(`err: post does not exist (${postPath})`);
	process.exit(1);
}

const postData = await getBlogpostData("private-blogposts", path.basename(postPath));
const postKey = crypto.randomUUID();

let keys = {
	[postKey]: {
		id: postData.id,
		createdAt: moment.now()
	}
}

const keysPath = "private-blogpost-keys.json";

if (fs.existsSync(keysPath)) {
	keys = {
		...keys,
		...JSON.parse(fs.readFileSync(keysPath).toString())
	};
}

fs.writeFileSync(keysPath, JSON.stringify(keys, null, 4));
console.log(`https://acikek.com/blog/${postData.id}/${postKey}`);

File: scripts/gen-private-post.py
#!/bin/python

import pathlib
import sys
import uuid
import json
import time

# todo: replace

if __name__ == '__main__':
	if len(sys.argv) < 2:
		print('err: provide blogpost name')
		exit(1)
	arg = str.join(' ', sys.argv[1:])
	path = pathlib.Path('private-blogposts') / arg
	path = path.with_suffix('.html')
	if not path.exists() or not path.is_file():
		print(f'err: post does not exist ({path})')
		exit(1)
	keys = { 
		str(uuid.uuid4()): {
			'file': str(path),
			'created-at': time.time()
		}
	}
	keys_path = pathlib.Path('private-blogpost-keys.json')
	if keys_path.exists():
		with open(keys_path, 'r') as keys_read:
			new_keys = json.load(keys_read)
			keys.update(new_keys)
	with open(keys_path, 'w') as keys_write:
		keys_write.write(json.dumps(keys, indent="    "))
